include/dlaf/matrix/internal/tile_future_manager.h

```diff
@@ -0,0 +1,79 @@
+//
+// Distributed Linear Algebra with Future (DLAF)
+//
+// Copyright (c) 2018-2019, ETH Zurich
+// All rights reserved.
+//
+// Please, refer to the LICENSE file in the root directory.
+// SPDX-License-Identifier: BSD-3-Clause
+//
+
+#include <hpx/hpx.hpp>
+#include "dlaf/tile.h"
+
+namespace dlaf {
+namespace matrix {
+namespace internal {
+
+template <class ReturnTileType, class TileType>
+hpx::future<ReturnTileType> setPromiseTileFuture(hpx::future<TileType>& old_future,
+                                                 hpx::promise<TileType>& p) noexcept {
```

Comment:
Since the promise `p` is an `In & move from` parameter, I would use an rvalue reference.

It is more informative from the user point of view since they must use `std::move` and they are aware that the promise will be moved and becomes invalid.

```suggestion
                                                 hpx::promise<TileType>&& p) noexcept {
```